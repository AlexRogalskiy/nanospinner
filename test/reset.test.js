delete process.env.CI

let { createSpinner } = require('../index.js')

let stdout = { out: '' }
stdout.write = symbols => {
  stdout.out += symbols
}

it(`spins default frames`, () => {
  let spinner = createSpinner('#reset', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.reset()
  spinner.spin()
  spinner.spin()

  let snapLocal = `"[?25l[1G[33mâ ‹[39m #reset[?25l[1G[33mâ ™[39m #reset[?25l[1G[33mâ ¹[39m #reset[?25l[1G[33mâ ‹[39m #reset[?25l[1G[33mâ ™[39m #reset"`
  let snapCI = `
    "[1G[33m-[39m #reset
    [1G[33m-[39m #reset
    [1G[33m-[39m #reset
    [1G[33m-[39m #reset
    [1G[33m-[39m #reset
    "
  `
  expect(stdout.out).toMatchInlineSnapshot(process.env.CI ? snapCI : snapLocal)
})
