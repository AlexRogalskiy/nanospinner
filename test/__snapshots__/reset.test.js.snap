// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spins default frames 1`] = `"[?25l[1G[33mâ ‹[39m #reset[?25l[1G[33mâ ™[39m #reset[?25l[1G[33mâ ¹[39m #reset[?25l[1G[33mâ ‹[39m #reset[?25l[1G[33mâ ™[39m #reset"`;
